tests/integration/testlib/vxlan.py

```diff
@@ -0,0 +1,87 @@
+#
+# Copyright (c) 2019 Red Hat, Inc.
+#
+# This file is part of nmstate
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+
+from contextlib import contextmanager
+
+import libnmstate
+
+from libnmstate.schema import Interface
+from libnmstate.schema import InterfaceState
+from libnmstate.schema import VXLAN
+
+
+class Vxlan:
+    def __init__(self, id, base_if, remote, destination_port=4789):
+        self.name = '{}.{}'.format(id, base_if)
+        self.id = id
+        self.base_if = base_if
+        self.remote = remote
+        self.destination_port = destination_port
+
+    def up_state(self):
+        return {
+            Interface.NAME: self.name,
+            Interface.TYPE: VXLAN.TYPE,
+            Interface.STATE: InterfaceState.UP,
+            VXLAN.CONFIG_SUBTREE: {
+                VXLAN.ID: self.id,
+                VXLAN.BASE_IFACE: self.base_if,
+                VXLAN.REMOTE: self.remote,
+                VXLAN.DESTINATION_PORT: self.destination_port,
+            },
+        }
+
+    def absent_state(self):
+        return {
+            Interface.NAME: self.name,
+            Interface.STATE: InterfaceState.ABSENT,
+        }
+
+    def down_state(self):
+        return {
+            Interface.NAME: self.name,
+            Interface.TYPE: VXLAN.TYPE,
+            Interface.STATE: InterfaceState.DOWN,
+        }
+
+
+@contextmanager
+def vxlan_interfaces(vxlans, create=True):
```

Comment:
Now, with python3, you can do this:
`def vxlan_interfaces(*vxlans, create=True):`

And in the caller you do not need the list/tuple:
`with vxlan_interfaces(Vxlan(...), Vxlan(...))`